src/libs/sfdk/virtualmachine.cpp

```diff
@@ -308,6 +320,13 @@ void VirtualMachine::setStorageSizeMb(int storageSizeMb, const QObject *context,
         const Functor<bool> &functor)
 {
     Q_D(VirtualMachine);
+    if (storageSizeMb > d->virtualMachineInfo.storageSizeMb) {
+        QTC_ASSERT(d->features & GrowStorageSize,
+                   QTimer::singleShot(0, context, std::bind(functor, false)); return);
+    } else if (storageSizeMb < d->virtualMachineInfo.storageSizeMb) {
+        QTC_ASSERT(d->features & ShrinkStorageSize,
+                   QTimer::singleShot(0, context, std::bind(functor, false)); return);
+    }
```

Comment:
I would like to handle also the case when no change is possible

```
QTC_ASSERT(d->features & (GrowStorageSize | ShrinkStorageSize),
        QTimer::singleShot(0, context, std::bind(functor, false)); return);
QTC_ASSERT(!(storageSizeMb > d->virtualMachineInfo.storageSizeMb) || d->features & GrowStorageSize,
        QTimer::singleShot(0, context, std::bind(functor, false)); return);
QTC_ASSERT(!(storageSizeMb < d->virtualMachineInfo.storageSizeMb) || d->features & ShrinkStorageSize,
        QTimer::singleShot(0, context, std::bind(functor, false)); return);
```

Reply:
Sure, if you want.

Reply:
But may I ask, why do you prefer !(storageSizeMb > d->virtualMachineInfo.storageSizeMb) over storageSizeMb <= d->virtualMachineInfo.storageSizeMb?

Reply:
yup, because (A -> B) == (!A || B). So whenever you see (!A || B) you know you can read it simply as "A implies B".